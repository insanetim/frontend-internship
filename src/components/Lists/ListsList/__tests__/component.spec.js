import { shallow } from 'enzyme'

import ListList from '../component'

const mockedHookData = {
  handlePaginationChange: jest.fn()
}
jest.mock('../hook', () => jest.fn(() => mockedHookData))

it('matches snapshot', () => {
  const mockedLists = {
    results: [
      {
        id: 1,
        title: 'test/title',
        overview: 'test/overview',
        poster_path: 'test/image'
      }
    ],
    total_pages: 10,
    total_results: 200
  }
  const component = shallow(<ListList lists={mockedLists} />)

  expect(component).toMatchSnapshot()
})

it('matches snapshot with 1 page', () => {
  const mockedLists = {
    results: [
      {
        id: 1,
        title: 'test/title',
        overview: 'test/overview',
        poster_path: 'test/image'
      }
    ],
    total_pages: 1,
    total_results: 20
  }
  const component = shallow(<ListList lists={mockedLists} />)

  expect(component).toMatchSnapshot()
})

it('matches snapshot without lists', () => {
  const mockedLists = {
    results: []
  }
  const component = shallow(<ListList lists={mockedLists} />)

  expect(component).toMatchSnapshot()
})
