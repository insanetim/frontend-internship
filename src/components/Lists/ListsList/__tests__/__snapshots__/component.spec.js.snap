// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <Row
    justify="center"
  >
    <Col
      span={20}
    >
      <Row
        gutter={
          {
            "lg": 32,
            "md": 24,
            "sm": 16,
            "xs": 8,
          }
        }
      >
        <ListItem
          key="1"
          list={
            {
              "id": 1,
              "overview": "test/overview",
              "poster_path": "test/image",
              "title": "test/title",
            }
          }
        />
      </Row>
    </Col>
  </Row>
  <Pagination
    onChange={[MockFunction]}
    pageSize={20}
    total={200}
  />
</Fragment>
`;

exports[`matches snapshot with 1 page 1`] = `
<Fragment>
  <Row
    justify="center"
  >
    <Col
      span={20}
    >
      <Row
        gutter={
          {
            "lg": 32,
            "md": 24,
            "sm": 16,
            "xs": 8,
          }
        }
      >
        <ListItem
          key="1"
          list={
            {
              "id": 1,
              "overview": "test/overview",
              "poster_path": "test/image",
              "title": "test/title",
            }
          }
        />
      </Row>
    </Col>
  </Row>
</Fragment>
`;

exports[`matches snapshot without lists 1`] = `
<Empty
  description="No results"
/>
`;
